projects/igniteui-angular/src/lib/time-picker/time-picker.directives.ts

```diff
@@ -127,8 +127,14 @@ export class IgxItemListDirective {
         const listName = (event.target as HTMLElement).className;
 
         if (listName.indexOf('hourList') !== -1) {
-            this.timePicker.minuteList.nativeElement.focus();
-        } else if (listName.indexOf('minuteList') !== -1 && this.timePicker._ampmItems.length !== 0) {
+            if (this.timePicker.minuteList) {
+                this.timePicker.minuteList.nativeElement.focus();
+            } else if (this.timePicker.ampmList) {
+                this.timePicker.ampmList.nativeElement.focus();
+            }
+        }
+
+        if (listName.indexOf('minuteList') !== -1 && this.timePicker.ampmList) {
             this.timePicker.ampmList.nativeElement.focus();
```

Comment:
duplicate code with the line above (ampmList.nativeElement.focus()) - merge if possible